### DISCUSSION TITLE
Why Pallene is not written in pallene?

### DISCUSSION THREAD

**Discussion by grifx** [pallene-lang/pallene#591] [Q&A] [answered]
Why Pallene is not written in pallene?

**Comment by hugomg** [comment, marked as answer]
We want to do that in the future but at the moment it's not possible. There are some features we would beed, that we still need to implement. For example nillable types and hash tables.

**Reply by grifx** [reply]
Is the grammar incomplete or are the underlying C codegen runtime utils not implemented?

Wouldn't it be beneficial to implement the LUA codegen for those features so that Pallene can be written in Pallene and compiled using --emit-lua?

**Reply by hugomg** [reply]
I think the main limitation at the moment is that the type system is not expressive enough to implement everything we do in the compiler. For example, we still need to design and implement nillable types and hash table types. The module system is also a big sticking point.

Converting the compiler to Pallene is a valuable long-term goal. I think the way to go would be to do it gradually. Pallene plays nice with Lua so we ought to be able to convert one file at a time.

**Reply by hugomg** [reply]
ps.: any help in implementing these things would always be appreciated ;)

**Reply by grifx** [reply]
Ok thanks, I thought --emit-lua was just removing the types, so you wouldn't have to do anything and just compile with --emit-lua.
